articles/service-fabric/how-to-managed-cluster-vmss-extension.md

```diff
@@ -55,6 +55,77 @@ Alternately, you can add a virtual machine scale set extension on a Service Fabr
+
+>[!NOTE]
+> It's essential to note that if an extension marked with `BeforeSFRuntime` fails, it will prevent the Service Fabric runtime from starting. Consequently, the node will be down from the Service Fabric perspective. Therefore, it is crucial to maintain these extensions with correct configurations and promptly address any issues that may arise to ensure the health of nodes within the cluster.
+
+### Requirements
+Use Service Fabric API version `2023-09-01-preview` or later.
+
+### ARM Template example:
+```json
+{
+  "type": "Microsoft.ServiceFabric/managedclusters/nodetypes",
+  "apiVersion": "2023-09-01-preview",
+  "name": "[concat(parameters('clusterName'), '/', parameters('nodeTypeName'))]",
+  "properties": {
+    "isPrimary": true,
+    ...
+    "vmExtensions": [
+        {
+            "name": "KvExtension",
+            "properties": {
+                "setupOrder": [
+                    "BeforeSFRuntime"
+                ],
+                "provisionAfterExtensions" [ "GenevaMonitoringExtension" ],
+                "publisher": "Microsoft.Azure.KeyVault",
+                "type": "KeyVaultForWindows",
+                "typeHandlerVersion": "3.0",
+                "autoUpgradeMinorVersion": true,
+                "settings": {
+                "secretsManagementSettings": {
+                  "observedCertificates": [
+                    ...
+                  ]
+                }
+              }
+            }
+        },
+        {
+          "name": "GenevaMonitoringExtension",
+          "properties": {
+              "setupOrder": [
+                    "BeforeSFRuntime"
+                ],
+              "autoUpgradeMinorVersion": true,
+              "enableAutomaticUpgrade": true,
+              "publisher": "Microsoft.Azure.Geneva",
+              "type": "GenevaMonitoring",
+              "typeHandlerVersion": "2.40",
+              "settings": {
+                "configurations": [
+                  {
+                    "ServiceArguments": {
+                      ...
+                    },
+                    "UserArguments": {
+                      ...
+                    }
+                  }
+                ]
+              }
+          }
+      }
+    ]
+  }
+}
+```
+
+>[!NOTE]
+> Special handling for AzureDiskEncryption (ADE) extension: ADE needs to run before the Service Fabric runtime to ensures that the disk is decrypted after a reimage operations, allowing the Service Fabric runtime to start using it. Even if the extension is not explicitly marked with `BeforeSFRuntime`, it will run before the runtime. But note that enabling encryption at host is recomended over using ADE extension. For detailed instructions, please refer to [Enable encryption at host](how-to-managed-cluster-enable-disk-encryption.md#enable-encryption-at-host)
```

Comment:
```suggestion
> Special handling for AzureDiskEncryption (ADE) extension: ADE needs to run before the Service Fabric runtime to ensures that the disk is decrypted after a reimage operations, allowing the Service Fabric runtime to start using it. Even if the extension is not explicitly marked with `BeforeSFRuntime`, it will run before the runtime. But note that enabling encryption at host is recomended over using ADE extension. For detailed instructions, please refer to [Enable encryption at host](how-to-managed-cluster-enable-disk-encryption.md#enable-encryption-at-host).

```